src/CMakeLists.txt

```diff
@@ -28,22 +35,55 @@ add_library(
   utility.cpp
 )
 
-target_link_libraries(insect_lib PUBLIC sfml-graphics)
+target_link_libraries(nectar_lib PUBLIC sfml-graphics)
 
 # Build simulator
 add_executable(
-  Insect_Sim 
+  nectar 
   main.cpp
   application.cpp
 )
-target_link_libraries(Insect_Sim PUBLIC insect_lib)
-#target_link_libraries(Insect_Sim PRIVATE sfml-graphics)
-target_compile_features(Insect_Sim PRIVATE cxx_std_17)
+
+target_link_libraries(nectar PUBLIC nectar_lib)
+target_compile_features(nectar PRIVATE cxx_std_17)
 
 # copy SFML dlls, for windows
 if (WIN32 AND BUILD_SHARED_LIBS)
-  add_custom_command(TARGET Insect_Sim POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_RUNTIME_DLLS:Insect_Sim> $<TARGET_FILE_DIR:Insect_Sim> COMMAND_EXPAND_LISTS)
+  add_custom_command(
+    TARGET nectar POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_RUNTIME_DLLS:nectar> $<TARGET_FILE_DIR:nectar> COMMAND_EXPAND_LISTS
+  )
 endif()
 
-install(TARGETS Insect_Sim)
+################################################################################
+# Doxygen documentation
+################################################################################
+
+option(BUILD_DOC "Build documentation with Doxygen" ON)
+
+find_package(Doxygen REQUIRED dot)
```

Comment:
Added dot requirement, because when I ran it before installing it, I'd get the "Doxygen build started" message before it'd fail